### PR TITLE
Add homework scores

### PR DESCRIPTION
**Pull request by beckybeauchamp1**
Needs Refactoring!

### CODE REVIEW

File: app/assets/javascripts/submission.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+$(".submission-score").on("keydown", function(evt){
```
Comment (jshawl) [inline app/assets/javascripts/submission.js]: feel like a `.on('submit')` is better / more flexible/semantic

File: app/controllers/submissions_controller.rb [hunk 2]
```diff
@@ -23,6 +23,13 @@ def destroy
     redirect_to @submission.assignment
   end
 
+  def update_score
+    submission = Submission.find(params[:submission_id])
+    submission.score = params[:score]
+    submission.save
```
Comment (jshawl) [inline app/controllers/submissions_controller.rb]: `.update(score: params[:score])`

Comment (mattscilipoti) [inline app/controllers/submissions_controller.rb]: Why isn't this just a submission update, where params contains `{submission: { score: 2 }}`?  Do we need another action?

Comment (andrewsunglaekim) [inline app/controllers/submissions_controller.rb]: resolved, changed ajax to path to existing update, thanks @mattscilipoti !

File: app/assets/javascripts/submission.js (hunk 1 above)
Comment (andrewsunglaekim) [inline app/assets/javascripts/submission.js]: can only do `submit` on form tag, we only have input, should we just add a form wrapper for the input tag?

Comment (RobertAKARobin) [inline app/assets/javascripts/submission.js]: Pretty sure "submit" is tricky and may vary a bit from browser to browser. I'd recommend sticking with `keydown`.

Comment (andrewsunglaekim) [inline app/assets/javascripts/submission.js]: refactored to submit! was a little tricky, can always switch back to keydown if neccessary

File: app/assets/javascripts/submission.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  var url = "../submissions/" + submissionId;
+  $.ajax({
+    url: url,
+    dataType: 'json',
+    type: "put",
+    data: {
+      score: inputEl.val()
+    }
+  }).done(function(res){
+    inputEl.parents("tr").append($("<td>Score Update</td>"))
```
Comment (joe-gz) [inline app/assets/javascripts/submission.js]: change parents to ".closest("tr")"

File: app/views/assignments/show.html.erb [hunk 4]
```diff
@@ -34,6 +34,7 @@
       <% Submission.statuses.keys.each do |status| %>
         <td><%= status_button(submission, status) %></td>
       <% end %>
+      <td><form class="score-update-form"><input class="submission-score" submission-id="<%=submission.id%>" type="text" value="<%=submission.score%>"></form></td>
```
Comment (adambray) [inline app/views/assignments/show.html.erb]: Can we make this `js-score-update-form`

**Pull request merged**